CRM/Event/BAO/Event.php

```diff
@@ -2042,13 +2052,18 @@ public static function checkPermission($eventId = NULL, $type = CRM_Core_Permiss
     static $permissions = NULL;
 
     if (empty($permissions)) {
-      $allEvents = CRM_Event_PseudoConstant::event(NULL, TRUE);
-      $createdEvents = array();
-
-      $session = CRM_Core_Session::singleton();
-      if ($userID = $session->get('userID')) {
-        $createdEvents = array_keys(CRM_Event_PseudoConstant::event(NULL, TRUE, "created_id={$userID}"));
-      }
+      $result = civicrm_api3('Event', 'get', array(
+        'check_permissions' => 1,
+        'return' => 'title',
+      ));
+      $allEvents = CRM_Utils_Array::collect('title', $result['values']);
+
+      $result = civicrm_api3('Event', 'get', array(
+        'check_permissions' => 1,
+        'return' => 'title',
+        'created_id' => 'user_contact_id',
```

Comment:
Build shows a break at this point - maybe all failing test needs `$this->createLoggedInUser();` in them to validate `'created_id' => 'user_contact_id'` param?